ports/raspberrypi/common-hal/nvm/ByteArray.c

```diff
@@ -71,6 +72,8 @@ void common_hal_nvm_bytearray_get_bytes(const nvm_bytearray_obj_t *self,
 
 bool common_hal_nvm_bytearray_set_bytes(const nvm_bytearray_obj_t *self,
     uint32_t start_index, uint8_t *values, uint32_t len) {
+    // disable interrupts to prevent core hang on rp2040
+    common_hal_mcu_disable_interrupts();
```

Comment:
The actual erasing and writing is done in `erase_and_write_sector()`, lines 64 and 65 above. I would suggest moving the disable/enable interrupts to wrap those two lines instead, since that's at the lowest level and will work for any use of that routine. You can see similar disable/enable code in `ports/raspberrypi/supervisor/internal_flash.c`, which does that wrapping around.

Reply:
implemented this change in the latest commit. Also had to add the disable / re-enable to `write_page()` since that gets called in the case where nvm was previously filled with `1`s  before writing. I confirmed that it could still hang when it had all `1`s in nvm before writing.

I tested the newest version successfully on Pimoroni PicoSystem